### DISCUSSION TITLE
Feature Request: Support for UPS batteries

### DISCUSSION THREAD

**Discussion by iridris** [andrew-codechimp/HA-Battery-Notes#645] [Ideas]
I have my UPS integrated with HA via the NUT add-on. My UPS has 2 batteries in it that need replaced every few years - it would be great to track these just like all the other battery-powered devices I have!

**Comment by andrew-codechimp** [comment]
I've seen a few UPS's added to the library that have a battery type of "Rechargeable" so you could always add the battery manually like that or am I missing something?

**Reply by iridris** [reply]
I do see that now, yes, so that's a possibility.

But I'm not 100% sure it should be listed as "Rechargeable". I mean it is rechargeable, but the batteries still have a lifespan and will require replacement every ~3 years. But the other complication is that the battery isn't a "standard" battery type like CR2032 or AA, it is a 12V sealed lead-acid battery with specific properties based on the UPS unit. 

If you can provide guidance on how to list that, I can submit the PR to add to the library.

**Reply by andrew-codechimp** [reply]
If it requires a specific type of battery you can add it via the [form](https://github.com/andrew-codechimp/HA-Battery-Notes/issues/new?template=new_device_request.yml&title=%5BDevice%5D%3A+), copy the exact manufacturer/model from the Device Info screen.

I think those specifying rechargeable are for sealed devices where it isn't replaceable.

You can then use the last replaced button/date manually as the time you replaced the battery rather than charged it up.

**Reply by iridris** [reply]
Thanks Andrew! Submitted the form, hopefully the naming is good enough.
https://github.com/andrew-codechimp/HA-Battery-Notes/issues/646

**Reply by andrew-codechimp** [reply]
That looks good to me, there's a different CPS UPS model in the library already so fingers crossed it will work.
